File: opencv/DisplayAVideo.cpp
/*
 * Write a program in both C++
 * to open and display the video stream from your webcam in a window.
 * Reading: OpenCV Documentation on VideoCapture
 *
*/


#include <opencv2/core.hpp>
#include <opencv2/videoio.hpp>
#include <opencv2/highgui.hpp>
#include <iostream>
#include <stdio.h>


int main()
{
	cv::Mat frame;
	cv::VideoCapture cap(0);
 
	if (!cap.isOpened()) {
		std::cerr << "ERROR! Unable to open camera" << std::endl;
		return -1;
 	}

	std::cout << "Press any key to terminate" << std::endl;
	
	while (true) {
		cap.read(frame);
		
		cv::imshow("Video", frame);
 		if (cv::waitKey(30) >= 0) {
 			break;
		}
 	}
	
	return 0;
}

File: opencv/DisplayAVideo.py
"""
  Write a program in both Python
  to open and display the video stream from your webcam in a window.
  Reading: OpenCV Documentation on VideoCapture
 
"""


import cv2

cap = cv2.VideoCapture(0);

if not cap.isOpened():
    print("ERROR")
    exit()

print("Press any key to terminate")

while True:
        ret, frame = cap.read()

        cv2.imshow('Video', frame)
        
        if cv2.waitKey(1) >= 0:
            break




